packages/ui-components/src/components/calendar-range-dates-selector/calendar-range-dates-selector.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core';
+import { fromDateToISO, fromISOToMoment, getDatesBetweenRange, isDateBefore, isDateSame } from '../../utils/date.helper';
+import { IClickDateEvent } from '../calendar/calendar.types';
+import { ICalendarRangeDatesSelector, ICalendarRangeDatesSelectorEvents, ISelectRangeDates } from './calendar-range-dates-selector.types';
+
+@Component({
+	tag: 'kv-calendar-range-dates-selector',
+	styleUrls: {
+		night: 'calendar-range-dates-selector.night.scss',
+		light: 'calendar-range-dates-selector.light.scss'
+	},
+	shadow: true
+})
+export class KvCalendarRangeDatesSelector implements ICalendarRangeDatesSelector, ICalendarRangeDatesSelectorEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: false }) selectedRangeDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) initialDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) disabledDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) minDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) maxDate?: string;
+
+	/** @inheritdoc */
+	@Event() selectRangeDates: EventEmitter<ISelectRangeDates>;
+
+	private onClickDate = ({ detail: { event, payload: date } }: CustomEvent<IClickDateEvent>): void => {
+		const clickedDateMoment = fromISOToMoment(date);
+
+		const [selectedStartDate, selectedEndDate] = this.getSelectedRangeDates();
+
+		// check if start date is not selected
```

Comment:
Would this be better to check with isEmpty?

Reply:
I don't see the need for that. 
Here I know that `selectedStartDate` is either undefined or an ISO formatted date string. 
The function `isEmpty` would do exactly what I'm doing here, no need for the function call overhead.